test/metabase/api/collection_test.clj

```diff
@@ -861,7 +862,18 @@
                                               :descrption "My SQL Snippets"
                                               :namespace  "snippets"})))
           (finally
-            (db/delete! Collection :name collection-name)))))))
+            (db/delete! Collection :name collection-name)))))
+    (testing "collection types"
+      (mt/with-model-cleanup [Collection]
+        (testing "Admins should be able to create with a type"
+          (mt/user-http-request :crowberto :post 200 "collection"
+                                {:name "foo", :color "#f38630", :type "official"})
+          (testing "But they have to be valid types"
+            (mt/user-http-request :crowberto :post 400 "collection"
+                                  {:name "foo", :color "#f38630", :type "no-way-this-is-valid-type"})))
+        (testing "Non-admins cannot create a collection with a type"
+          (mt/user-http-request :rasta :post 403 "collection"
+                                {:name "foo", :color "#f38630", :type "official"}))))))
```

Comment:
You should consider adding an assertion for the actual API response here as well to make sure it returns the correct error in the shape we expect